### PR TITLE
STONEBLD-1494: Add procedure to update buildah task

### PR DESCRIPTION
**Pull request by Michkov**
To unblock building applications with higher memory requirements we are providing new buildah tasks which can be used by user.

Also included commit with update of taskRef. Tekton/Openshift Pipelines are supporting multiple resolvers now, which  changed format of taskRef definition. Old way is still possible but having performace issues.

### CODE REVIEW

**Comment by gbenhaim** [comment]
lgtm

**Review by gbenhaim — APPROVED** [review]

**Review by gabemontero — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 1]
```diff
@@ -62,6 +68,19 @@ NOTE: If you want to define a task directly in this file, rather than using `tas
 +
 . Commit your changes to the repository of the component.
 
+== Exchanging the build pipeline build task with higher memory limits
+
+Task `buildah` for building components from Dockerfile has a memory limit set to 4 GB. For components with higher memory requirements, there are tasks prepared:
```
Comment (Chr1st1anSears) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:73]: Suggested rewrite: 
The `buildah` task, which builds components from a Dockerfile, has a memory limit of 4 GB. To build components with memory requirements greater than 4 GB, use the following tasks:

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 2]
```diff
@@ -62,6 +68,19 @@ NOTE: If you want to define a task directly in this file, rather than using `tas
...
+
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-6gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb]
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-8gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-8gb]
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-10gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-10gb]
+
+.Procedure
+
+. In each `.yaml` file in the `.tekton` directory, locate a task named `build-container` under `tasks`:
+.. Set under `.taskRef.params` `name` to `buildah-6gb`
```
Comment (Chr1st1anSears) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:82]: Suggested rewrite:
..Under `.taskRef.params`, set `name` to `buildah-6gb`.

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 3]
```diff
@@ -62,6 +68,19 @@ NOTE: If you want to define a task directly in this file, rather than using `tas
...
+.. Set under `.taskRef.params` `bundle` to `quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb:0.1`
```
Comment (Chr1st1anSears) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:83]: Suggested rewrite:
..Under `.taskRef.params`, set `bundle` to `quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb:0.1`.

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 4]
```diff
@@ -62,6 +68,19 @@ NOTE: If you want to define a task directly in this file, rather than using `tas
...
+
+Task `buildah` for building components from Dockerfile has a memory limit set to 4 GB. For components with higher memory requirements, there are tasks prepared:
+
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-6gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb]
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-8gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-8gb]
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-10gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-10gb]
+
+.Procedure
+
+. In each `.yaml` file in the `.tekton` directory, locate a task named `build-container` under `tasks`:
```
Comment (Chr1st1anSears) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:81]: Suggested rewrite:
. Go to the GitHub repo of your component.
. In each `.yaml` file in the `.tekton` directory, under `tasks`, locate the task named `build-container`:

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 5]
```diff
@@ -62,6 +68,19 @@ NOTE: If you want to define a task directly in this file, rather than using `tas
...
+== Exchanging the build pipeline build task with higher memory limits
+
+Task `buildah` for building components from Dockerfile has a memory limit set to 4 GB. For components with higher memory requirements, there are tasks prepared:
+
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-6gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb]
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-8gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-8gb]
+* link:https://quay.io/repository/redhat-appstudio-tekton-catalog/task-buildah-10gb?tab=tags[quay.io/redhat-appstudio-tekton-catalog/task-buildah-10gb]
+
+.Procedure
+
```
Comment (Chr1st1anSears) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:80]: Suggested addition:
To exchange the build task with a memory limit of 6 GB, complete the following steps. For a memory limit of 8 or 10 GB, replace the references to 6 GB with the appropriate values.

**Review by Chr1st1anSears — CHANGES_REQUESTED** [review]

**Comment by Chr1st1anSears** [comment]
Hey @Michkov--thanks for your PR and for requesting my review! I appreciate the helpful updates you've provided to the technical details of our doc. However, to ensure that the doc complies with our best practices as writers, I've requested a number of changes. Please let me know if you have any questions about them.

**Comment by Michkov** [comment]
@Chr1st1anSears thanks for review. Document updated.

BTW github provides ability to directly propose the changes of the code. Then I would be able to just accept your suggestions ;-).
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request - point 6

**Comment by Chr1st1anSears** [comment]
@Michkov looks good and you're right lol that would definitely be a better way to leave a review next time. Thanks for pointing that out.

**Review by Chr1st1anSears — APPROVED** [review]

**Pull request merged**